mpd/duration.go

```diff
@@ -141,77 +143,61 @@ func fmtInt(buf []byte, v uint64) int {
 
 func parseDuration(str string) (time.Duration, error) {
 	if len(str) < 3 {
-		return 0, errors.New("input duration too short")
+		return 0, errors.New("At least one number and designator are required")
 	}
 
-	var minus bool
-	offset := 0
-	if str[offset] == '-' {
-		minus = true
-		offset++
+	if strings.Contains(str, "-") {
+		return 0, errors.New("Duration cannot be negative")
 	}
 
-	if str[offset] != 'P' {
-		return 0, errors.New("input duration does not have a valid prefix")
+	// Check that only the parts we expect exist and that everything's in the correct order
+	r := regexp.MustCompile(
```

Comment:
Can we move the regex initialisation outside the method? Compiling the regex on every invocation feels unnecessary.

Reply:
Yep, done!